### PR TITLE
ORC-907: Remove junit-vintage-engine from core module

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR removes `junit-vintage-engine`.

### Why are the changes needed?

After this PR, Apache ORC JUnit5 migration is finished.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

File: java/pom.xml [hunk 1]
```diff
@@ -779,28 +779,22 @@
...
       <dependency>
         <groupId>org.junit.jupiter</groupId>
         <artifactId>junit-jupiter-params</artifactId>
         <version>5.7.0</version>
         <scope>test</scope>
       </dependency>
       <dependency>
-        <groupId>org.junit.vintage</groupId>
-        <artifactId>junit-vintage-engine</artifactId>
-        <version>5.7.2</version>
```
Comment (dongjoon-hyun) [inline java/pom.xml:797]: This is the last PR to remove `vintage` engine completely.

**Comment by dongjoon-hyun** [comment]
It's `findbags` failures.
```
[INFO] --- findbugs-maven-plugin:3.0.5:check (analyze-compile) @ orc-core ---
[INFO] BugInstance size is 2
[INFO] Error size is 0
[INFO] Total bugs: 2
[INFO] Comparison of String parameter using == or != in org.apache.orc.TestUnicode.testUtf8(String, int, boolean)  [org.apache.orc.TestUnicode] At TestUnicode.java:[line 77] ES_COMPARING_PARAMETER_STRING_WITH_EQ
[INFO] Call to equals(null) in org.apache.orc.impl.TestStreamName.test1() [org.apache.orc.impl.TestStreamName] At TestStreamName.java:[line 41] EC_NULL_ARG
```

**Comment by dongjoon-hyun** [comment]
cc @williamhyun and @pgaref

**Comment by dongjoon-hyun** [comment]
This is the final piece of JUnit5 migration for 1.7.0 release. I hope `main` and `branch-1.7` similarly because this is a big change.

**Comment by dongjoon-hyun** [comment]
Could you review this too, please, @pgaref ?
I know that this is a heavy PR, but most changes are straightforward replacement.

**Comment by dongjoon-hyun** [comment]
Could you review this please, @williamhyun ?

**Review by williamhyun — APPROVED** [review]
+1, There was a lot of the same change throughout the files, it was tough to review but it looks like it's going in the right direction.

**Comment by dongjoon-hyun** [comment]
Thank you so much, @williamhyun . Ya, I know that this is too big and tedious PR. :)
Merged to main/1.7.

**Comment by dongjoon-hyun** [comment]
This is backported to branch-1.7 with #809

**Pull request merged**